### PR TITLE
Enable "when" for workflow steps

### PR DESCRIPTION
**Pull request by mr-c**
- [x] rebuild java classes

### CODE REVIEW

File: java/src/main/java/org/galaxyproject/gxformat2/v19_09/WorkflowStep.java [hunk 1]
```diff
@@ -205,4 +219,17 @@
...
+   * <BLOCKQUOTE>
+   *
+   * If defined, only run the step when the expression evaluates to `true`. If `false` the step is
+   * skipped. A skipped step produces a `null` on each output.
+   *
+   * <p>Expression should be an ecma5.1 expression. *
+   *
+   * </BLOCKQUOTE>
+   */
+  java.util.Optional<String> getWhen();
```
Comment (mr-c) [inline java/src/main/java/org/galaxyproject/gxformat2/v19_09/WorkflowStep.java:234]: Hello!

File: schema/v19_09/workflow.yml [hunk 2]
```diff
@@ -226,19 +226,19 @@ $graph:
...
+      type:
+        - "null"
+        - string  # TODO: cwl defines an enum for this, not sure how that works
+      jsonldPredicate:
+        _id: "gxformat2:when"
+        noLinkCheck: true
+      doc: |
+        If defined, only run the step when the expression evaluates to
+        `true`.  If `false` the step is skipped.  A skipped step
+        produces a `null` on each output.
```
Comment (mr-c) [inline schema/v19_09/workflow.yml:239]: Hello!

File: gxformat2/schema/v19_09.py [hunk 3]
```diff
@@ -1910,49 +2093,59 @@ def save(self, top=False, base_url="", relative_uris=True):
...
+        tool_version: Optional[Any] = None,
+        errors: Optional[Any] = None,
+        uuid: Optional[Any] = None,
+        in_: Optional[Any] = None,
+        state: Optional[Any] = None,
+        tool_state: Optional[Any] = None,
+        type: Optional[Any] = None,
+        run: Optional[Any] = None,
+        runtime_inputs: Optional[Any] = None,
+        when: Optional[Any] = None,
```
Comment (mr-c) [inline gxformat2/schema/v19_09.py:2145]: Hi!

**Review by mr-c — COMMENTED** [review]
comments left to show off the new bits

**Comment by mr-c** [comment]
@denis-yuen  ; this PR updates the java bindings. Maybe you could test it for us?

File: java/src/main/java/org/galaxyproject/gxformat2/v19_09/Any.java [hunk 4]
```diff
@@ -1,3 +1,17 @@
+// Copyright Common Workflow Language project contributors
```
Comment (denis-yuen) [inline java/src/main/java/org/galaxyproject/gxformat2/v19_09/Any.java:1]: Think there's some inappropriate automatic code licensing going on, looks like license should be MIT
https://github.com/galaxyproject/gxformat2/blob/master/LICENSE unless I misunderstand what's going on

**Review by denis-yuen — COMMENTED** [review]
Hi, 
Just trying to get my bearings here. So it looks like the version of the format in the pom.xml is  v19_09  But the change is being proposed for 0.16.0.dev0 

What's the context for the addition of "when", is there a discussion elsewhere that I can read and get caught up on?

**Comment by mvdbeek** [comment]
Thanks for the review @denis-yuen,

> Just trying to get my bearings here. So it looks like the version of the format in the pom.xml is v19_09 But the change is being proposed for 0.16.0.dev0

0.16.0 will be the next the version of gxformat2, which contains multiple packages in different languages, `v19_09` corresponds to the oldest Galaxy version that supports all features of the gxformat2 schema and is also the version of the schema. We should probably create a new version with each Galaxy release (assuming new gxformat2 features were added), but we've been adding both new features (that were not yet available in Galaxy 19.09), and older, previously unsupported-in-gxformat2 features ... so it is a little messy. Unfortunately I don't think we have the setup to create a new version. Correct me if I'm wrong please @jmchilton.

**Comment by mvdbeek** [comment]
> What's the context for the addition of "when", is there a discussion elsewhere that I can read and get caught up on?

A `when` on a workflow step will result in conditional execution of that step. We haven't added this to core Galaxy yet, but this is in preparation of doing that. The WIP commit that adds the missing piece is in https://github.com/common-workflow-lab/galaxy/pull/123/commits/7412df6a9bb00d24b53a0a83815929c04c30c5e4.

**Review by denis-yuen — APPROVED** [review]
Passed tests on our side, see linked PR for how we went about it (albeit with a big lag)

**Comment by mvdbeek** [comment]
Thank you @mr-c!

**Pull request merged**